src/buildercore/core.py

```diff
@@ -233,26 +260,35 @@ def describe_stack(stackname):
     return first(connect_aws_with_stack(stackname, 'cfn').describe_stacks(stackname))
 
 # TODO: rename or something
-def stack_data(stackname):
+def stack_data(stackname, ensure_single_instance=False):
     """like `describe_stack`, but returns a dictionary with the Cloudformation 'outputs' 
-    indexed by key and ec2 data under the key 'instance'"""
+    indexed by key and ec2 data under the key 'instance'
+    
+    Returns a list if more than one result is found, but otherwise sticks
+    to a single dictionary for backward compatibility"""
     stack = describe_stack(stackname)
-    data = stack.__dict__
-    if data.has_key('outputs'):
-        data['indexed_output'] = {row.key: row.value for row in data['outputs']}
     try:
         # TODO: is there someway to go straight to the instance ID ?
         # a CloudFormation's outputs go stale! because we can't trust the data it
         # gives us, we sometimes take it's instance-id and talk to the instance directly.
-        #inst_id = data['indexed_output']['InstanceId']
-        #inst = get_instance(inst_id)
-        stacks = find_ec2_instance(stackname)
-        assert len(stacks) == 1, ("while looking for %s, found these stacks: %s" % (stackname, stacks))
-        inst = stacks[0]
-        data['instance'] = inst.__dict__
+        ec2_instances = find_ec2_instance(stackname)
+        assert len(ec2_instances) >= 1, ("while looking for %s, found no ec2 instances" % stackname)
+        if len(ec2_instances) == 1:
```

Comment:
`find_ec2_instance` should already do that, but it is this method that has the two possible cases (dictionary for one, or list of dictionaries if more than one); that's the reason for the two branches. But if *this* method always returns a list, then it can be unified